UnitTests/DeepLinkParserTests.cs

```diff
@@ -0,0 +1,202 @@
+﻿using Microsoft.Toolkit.Uwp;
+using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;
+using System;
+using System.Diagnostics;
+using UnitTests;
+
+namespace Tests
+{
+    [TestClass]
+    public class DeepLinkParserTests
+    {
+        public TestContext TestContext { get; set; }
+
+        private const string SAMPLELINK = @"MainPage/Options?option1=value1&option2=value2&option3=value3";
+        private static readonly DeepLinkParser _parser = new TestDeepLinkParser(SAMPLELINK);
+
+        [TestMethod]
+        public void Test_DeepLink_RootValue()
+        {
+            Assert.AreEqual("MainPage/Options", _parser.Root);
+        }
+
+        [TestMethod]
+        public void Test_DeepLink_PullOptions()
+        {
+            Assert.AreEqual("value1", _parser["option1"]);
+            Assert.AreEqual("value2", _parser["option2"]);
+            Assert.AreEqual("value3", _parser["option3"]);
+        }
+
+        [TestMethod]
+        public void Test_DeepLink_OptionNotFound()
+        {
+            try
+            {
+                var nonexistentvalue = _parser["nonexistentoption"];
+
+                Assert.Fail("Should have thrown KeyNotFoundException");
+            }
+            catch (System.Collections.Generic.KeyNotFoundException)
+            {
+            }
+            catch
+            {
+                Assert.Fail("Should have thrown KeyNotFoundException");
+            }
+        }
+
+        [TestMethod]
+        public void Test_DeepLink_DuplicateKeys()
```

Comment:
I think the parser shouldn't throw an exception because of a duplicate key, in asp.net if there are 2 queryStrings keys with same name and you tried to get the value of this duplicate key it will provide both values comma separated ex:
"value1,value2"=QueryString["duplicateKey"]
@ScottIsAFool @deltakosh what do you think ?